src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1525,6 +1527,14 @@ private function dumpValue($value, $interpolate = true)
             }
 
             return sprintf('array(%s)', implode(', ', $code));
+        } elseif ($value instanceof LocatorArgument) {
+            $code = "\n";
+            foreach ($value->getValues() as $k => $v) {
+                $code .= sprintf("            '%s' => function () { return %s; },\n", $k, $v instanceof Reference ? $this->getServiceCall((string) $v, $v) : $v);
```

Comment:
`$this->dumpValue($v, $interpolate)` instead of the ternary op
`$this->dumpValue($k, $interpolate)` for the key